# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: uriloader/exthandler/uikit/nsOSHelperAppService.mm
# Commit: f0094c9184c4
# Full Hash: f0094c9184c41675f11efc2d4332ec1a63e570bc
# Author: Gijs Kruitbosch <gijskruitbosch@gmail.com>
# Date: 2019-11-27 09:44:05
# Regressor Bug: 1496380
# File Overlap Count: 1
# Description:
#   Bug 1496380 - stop recursion via the external protocol handler if Firefox is either the default OS handler or a configured external handler, r=mossop
#   
#   This is an initial implementation of this idea that works on mac.
#   I've added a Windows implementation in another commit in this stack. I'll
#   look at a Linux one in a follow-up bug. I do not think we need them in the
# ==============================================================================

diff -r 9dbf3a2c8c37 -r f0094c9184c4 uriloader/exthandler/uikit/nsOSHelperAppService.mm
--- a/uriloader/exthandler/uikit/nsOSHelperAppService.mm	Tue Nov 26 18:47:20 2019 +0000
+++ b/uriloader/exthandler/uikit/nsOSHelperAppService.mm	Tue Nov 26 18:51:04 2019 +0000
@@ -21,6 +21,11 @@
   return NS_ERROR_NOT_AVAILABLE;
 }
 
+NS_IMETHODIMP
+nsOSHelperAppService::IsCurrentAppOSDefaultForProtocol(const nsACString& aScheme, bool* _retval) {
+  return NS_ERROR_NOT_AVAILABLE;
+}
+
 nsresult nsOSHelperAppService::GetFileTokenForPath(const char16_t* aPlatformAppPath,
                                                    nsIFile** aFile) {
   return NS_ERROR_NOT_IMPLEMENTED;